perceval/converters/cqasm_converter.py

```diff
@@ -0,0 +1,331 @@
+        if num_controls >= 2:
+            raise ConversionUnsupportedFeatureError(
+                f"Gate { gate_name } has more than one control (n = { num_controls })")
+
+        if not controls:
+            circuit_template = _CQASM_1_QUBIT_GATES.get(gate_name, None)
+            if not circuit_template:
+                raise ConversionUnsupportedFeatureError(
+                    f"Unsupported 1-qubit gate { gate_name }")
+            if parameter:
+                circuit_template = circuit_template(parameter)
+            for target in targets:
+                circuit = circuit_template.copy()
+                self._converted_processor.add(target * 2, circuit.copy())
+        else:
+            if gate_name not in _CQASM_2_QUBIT_GATES:
+                raise ConversionUnsupportedFeatureError(
+                    f"Unsupported 2-qubit gate { gate_name }")
+            for target in targets:
+                self._create_2_qubit_gates_from_catalog(
+                    gate_name,
+                    self._num_cnots,
+                    controls[0] * 2,
+                    target * 2,
+                    min(target, controls[0]) * 2,
+                    self._use_postselection)
+
+    def convert(self, ast, use_postselection: bool = True) -> Processor:
+        r"""Convert a cQASM quantum program into a `Processor`.
+
+        :param ast: the AST of a cQASM program
+        :type ast: a Program object, as returned by the cQASM parser
+        :param use_postselection: when True, uses a `postprocessed CNOT`
+        as the last gate. Otherwise, uses only `heralded CNOT`
+        :return: the converted processor
+        """
+        if not type(ast) is cqasm.semantic.Program:
+            raise ConversionSyntaxError(f"cQASM parser error: { ast[0] }")
+
+        self._collect_qubit_list(ast)
+        self._num_cnots = sum(
+            (_cs(s.name) == "CNOT") + 2 * (_cs(s.name) in ["CR", "CRk"])
+            for s in ast.block.statements)
+        self._use_postselection = use_postselection
+
+        qubit_names = [
+            f'{ q }[{ i }]' if i >= 0 else q for (q, i) in self._qubit_list]
+        self._configure_processor(ast, qubit_names=qubit_names)
+        for statement in ast.block.statements:
+            self._convert_statement(statement)
+        self.apply_input_state()
+        return self._converted_processor
+        
+    def convert_string(
+            self,
+            source_string: str,
+            use_postselection: bool = True) -> Processor:
+        r"""Convert a cQASM quantum program into a `Processor`.
+
+        :param source_string: The cQASM program stored in a string
+        :param use_postselection: when True, uses a `postprocessed CNOT`
+        as the last gate. Otherwise, uses only `heralded CNOT`
+        :return: the converted processor
+        """
+        ast = cqasm.Analyzer().analyze_string(source_string)
+        return self.convert(ast, use_postselection)
+
+    def convert_file(
```

Comment:
There are multiple `convert` methods for each type of input. I would suggest refactoring the code to use `@multipledispatch` to distinguish between `str`, `cqasm` object. A filename which would be of type `str` to would require a check within that method to differentiate between filename and a string of instructions. Suggested check: a filename would not have a `newline` character in it.

This would clean up the code requiring the user to use the exact syntax as other converters of perceval `CQasmConverterObj.convert(qasm, use_postselection)`